src/main/java/io/github/syst3ms/skriptparser/expressions/ExecExprListOperators.java

```diff
@@ -0,0 +1,151 @@
+package io.github.syst3ms.skriptparser.expressions;
+
+import io.github.syst3ms.skriptparser.Parser;
+import io.github.syst3ms.skriptparser.lang.Expression;
+import io.github.syst3ms.skriptparser.lang.TriggerContext;
+import io.github.syst3ms.skriptparser.lang.base.ExecutableExpression;
+import io.github.syst3ms.skriptparser.log.ErrorType;
+import io.github.syst3ms.skriptparser.parsing.ParseContext;
+import io.github.syst3ms.skriptparser.types.changers.ChangeMode;
+import org.jetbrains.annotations.Nullable;
+
+import java.math.BigInteger;
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+/**
+ * Basic list operators with the following behavior:
+ * <ul>
+ *      <li>{@code pop} removes the last element and returns that removed element.</li>
+ *      <li>{@code shift/poll} removes the first element and returns that removed element.</li>
+ *      <li>{@code splice} removes elements in a certain bound and returns those removed elements.</li>
+ * </ul>
```

Comment:
Alright, I don't like how these syntaxes use programming jargon like pop, poll and splice. The goal is to be transparently understandable by non-programmers here. 
I suggest the word "extract" as an alternative to the first two syntaxes, with ways to specify the first, the last, or the n-th element.
As for "splice", it's a bit more understandable, but I don't think it should be mandatory. I suggest the following syntax:
```autohotkey
[splice] %objects% (1:from %integer% to %integer%|2:starting (at|from) %integer%|3:up to %integer%) [[with] step %integer%]
```
It has all the possibilities of the current syntax, and then some, bringing it a bit closer to Python splicing (the goal isn't to reach that exactly, we don't need things like `list[::-1]`).

EDIT : pop/poll/extract are redundant when used as effects, a proper "remove"/"delete" should be plenty (and less obscure), and perhaps it should be mandatory to use the "splice" part when using it as an effect.

Reply:
> EDIT : pop/poll/extract are redundant when used as effects, a proper "remove"/"delete" should be plenty (and less obscure), and perhaps it should be mandatory to use the "splice" part when using it as an effect.
Allowing them as an effect would allow people to write `pop {list::*}` instead of `remove last element of {list::*} from {list::*}`, which is a good benefit in my opinion. At least good enough to keep the syntax as an effect.

I'll add the other suggestions though since I do agree this syntax is a bit too clunky in terms of wording. 
Should I remove the whole `pop/shift` syntax or can it remain as an alternative to the new `extract` one?

Reply:
Sounds good, we let programmers do their programmer shenanigans, and let non-programmers do the easier stuff.